### PR TITLE
Inline childFromLineNumber and childFromCharOffset

### PR DESCRIPTION
**Pull request by andy-ms**
Should wait on #17017 and be rebased once that's in.
These functions were only used in one place, and the code is simpler if we just directly do the work instead of using two separate functions.

### CODE REVIEW

**Comment by andy-ms** [comment]
@sheetalkamat PR is ready now.

File: src/server/scriptVersionCache.ts [hunk 1]
```diff
@@ -672,85 +672,46 @@ namespace ts.server {
...
-                };
-            }
-            else if (childInfo.childIndex < this.children.length) {
-                if (childInfo.child.isLeaf()) {
-                    return {
-                        oneBasedLine: childInfo.lineNumberAccumulator,
-                        zeroBasedColumn: childInfo.relativePosition,
-                        lineText: childInfo.child.text,
-                    };
+            Debug.assert(this.children.length !== 0);
```
Comment (rbuckton) [inline src/server/scriptVersionCache.ts]: Is it guaranteed that `children` will _never_ be empty when this is called? The previous logic had a specific case to handle that (returning `child` as `undefined` from `childFromCharOffset`).

Comment (andy-ms) [inline src/server/scriptVersionCache.ts]: It looks like that is possible for a LineNode to have 0 children if this document is the empty string. Added a test.

**Comment by rbuckton** [comment]
Your test is failing.

**Comment by andy-ms** [comment]
Fixed.

**Pull request merged**